### PR TITLE
Profile sample uploading

### PR DESCRIPTION
**Pull request by rish9511**
Reasons that led to slow down of sample uploading:
1. enumerateSRMScan() and  calculateMzRtRange() were called thrice for every sample
2. Ofast and ffast-math flag were removed and hence compiler was not optimizing parts of code involved in sample uploading process

Solution
1. removed duplicated calls 
2. added compiler flags for windows and linux

Tested on:
Mac
Linux
Windows

### CODE REVIEW

File: src/core/libmaven/mzSample.cpp [hunk 1]
```diff
@@ -305,7 +305,7 @@ void mzSample::parseMzML(const char *filename)
 	xml_node experimentRun = doc.first_child().first_element_by_path("mzML/run");
 	if (!experimentRun.empty())
 	{
-		parseMzMLInjectionTimeStamp(experimentRun);
+        parseMzMLInjectionTimeStamp(experimentRun.attribute("startTimeStamp"));
```
Comment (saifulbkhan) [inline src/core/libmaven/mzSample.cpp]: Please correct indentation

Comment (saifulbkhan) [reply src/core/libmaven/mzSample.cpp]: Please ignore this. This appears totally fine in the editor.

File: src/core/libmaven/mzSample.cpp [hunk 2]
```diff
@@ -899,7 +898,7 @@ void mzSample::populateFilterline(string filterLine, Scan *_scan)
 	}
 }
 
-void mzSample::parseMzXMLScan(const xml_node &scan, int scannum)
+void mzSample::parseMzXMLScan(const xml_node &scan, const int& scannum)
```
Comment (shubhra-agrawal) [inline src/core/libmaven/mzSample.cpp]: @rish9511 Why pass int value as a const ref? pass-by-value would surely be same or faster in this case. Not really an issue, just curious.

Comment (rish9511) [reply src/core/libmaven/mzSample.cpp]: @shubhra-agrawal  
references make the code memory efficient not necessarily faster(unless it's a really large object)
if the parameters are just being read, they should be received as const references

Comment (shubhra-agrawal) [reply src/core/libmaven/mzSample.cpp]: I get that. I was reading about passing by const ref recently and most sources were divided on what is better for native datatypes: sending by value or const ref. Just wanted to know why you chose one over the other. Seems you're going with convention here, which is good. Thanks!

File: src/core/libmaven/mzSample.h [hunk 3]
```diff
@@ -706,7 +706,7 @@ class mzSample
 
     void setInstrumentSettigs(xml_document &doc, xml_node spectrumstore);
 
-    void parseMzXMLData(xml_document &staticdoc, xml_node spectrumstore);
+    void parseMzXMLData(const xml_node& spectrumstore);
```
Comment (shubhra-agrawal) [inline src/core/libmaven/mzSample.h]: Good eye! :)

File: src/gui/mzroll/mzfileio.cpp [hunk 4]
```diff
@@ -583,16 +574,8 @@ void mzFileIO::fileImport(void) {
             QString filename = samples.at(i);
             mzSample* sample = loadSample(filename);
             if (sample) {
-                    sample->enumerateSRMScans();
-                    sample->calculateMzRtRange();    //set min and max values for rt
-                    sample->fileName = filename.toStdString();
-
-                    if ( filename.contains("blank",Qt::CaseInsensitive))
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/mzfileio.cpp]: @rish9511 This should be removed from the else block as well, then

Comment (rish9511) [reply src/gui/mzroll/mzfileio.cpp]: @shubhra-agrawal 
Thanks for pointing out

**Pull request merged**